### PR TITLE
feat/third-lecture

### PR DESCRIPTION
**Pull request by niekstrv**
## :memo: Changelog

- [x] the notebook for the third lecture is added. 
- [x] I also added a few slides in order to explain the theoretical foundations better.

## :link: Links

- https://github.com/strvcom/ds-academy/issues/13

## :white_check_mark: Checklist

- [x] Lint and tests pass locally with my changes

## Extra comments
This lecture will require some extra theoretical foundations and I plan to do that with the slides included in the PR. Let me know what you think about having slides + a notebook ( I know that Honza disagrees :-P).

Please let me know if anything related to the discussed algorithms is sharply missing.

Also, in our google docs (https://docs.google.com/document/d/1y70tjwpTyntWU_-cbHKC4GGYapBZ6hHYOcWffInp2yw/edit) we also had dimensionality reduction planned for this lecture. However, I think that it is too much to include it here. I am expecting some question during the presentation so I wouldnt know how far I would get time-wise. I think that linear regression, logistic regression and clustering are a lot take in during a single lecture.

### CODE REVIEW

**Comment by honzaMaly** [comment]
Thanks for the work! A few points:

- Could you please merge Conda ENV with the primary environment? BTW - it seems that all those dependencies are already included. Dependencies should also be present in the DEV ENV so that the handbook can be generated.
- I would rephrase the lecture a bit - it seems that the main focus is linear/logistic regression. However, we should be a bit more general - the topic is classification/regression. Could we please put more emphasis on this fact? And use linear/logistic regression to explain those concepts? Because those are just algorithms to solve regression/classification...
- I don't particularly appreciate that there are separate slides. It would be great to have them in the notebook.
- Could you spend some time explaining concepts? It would be great to have some examples as well. Take this one as an example: https://jakevdp.github.io/PythonDataScienceHandbook/05.01-what-is-machine-learning.html. As a beginner, I would be lost.

**Review by jardabezdek — CHANGES_REQUESTED** [review]
#### 👍 

- I really like you have added a presentation to back up the theory behind the topic! 💪
- I appreciate you have come up with exercises on regression and clustering tasks using one data set. It is really cool! ❤️ 
- Exercises section in general!

#### 👎 

- I would suggest to send the presentation slides to Mrs. Linda Krestanova on review in order to push it to the next level. When I am thinking about this, maybe it would be good idea to send her the whole handbook before academy starts, so we are sure everything is as it should be. What do you think, @honzaMaly? 🤔 
- If I am not mistaken, you need to add your lecture structure into `_toc.yml` file in the repository root folder.
- In notebook in the cell 7, there is this code: `linear_regression_model = LinearRegression(fit_intercept=True).fit(X=train_linear_df['X'].values.reshape(-1, 1), y=train_linear_df['y'])`. Is it correct? I would say it should look like this: `linear_regression_model = LinearRegression(fit_intercept=True).fit(X=train_linear_df['X']`, but maybe I am missing something. The same pattern is in logistic regression section.
- When you finish your work on notebook, I would suggest to restart kernel and run all cells, so the numbers of cells are ordered. It is also a check for you that notebook can be re-run by other user without any errors.
- In the "EXERCISES 1B" section, there is written the following: `1b) Use sklearn's linear regression model on the non-zero elements of the 'age' feature to predict the zero-elements of the 'age' feature.` However, I think you meant something like this: `1b) Train sklearn's linear regression model on the non-zero elements of the 'age' feature to predict the missing values of the 'age' feature.`
- Links section in PR description does not make sense to me.

**Comment by honzaMaly** [comment]
> #### 👍
> * I really like you have added a presentation to back up the theory behind the topic! 💪
> * I appreciate you have come up with exercises on regression and clustering tasks using one data set. It is really cool! ❤️
> * Exercises section in general!
> 
> #### 👎
> * I would suggest to send the presentation slides to Mrs. Linda Krestanova on review in order to push it to the next level. When I am thinking about this, maybe it would be good idea to send her the whole handbook before academy starts, so we are sure everything is as it should be. What do you think, @honzaMaly? 🤔
> * If I am not mistaken, you need to add your lecture structure into `_toc.yml` file in the repository root folder.
> * In notebook in the cell 7, there is this code: `linear_regression_model = LinearRegression(fit_intercept=True).fit(X=train_linear_df['X'].values.reshape(-1, 1), y=train_linear_df['y'])`. Is it correct? I would say it should look like this: `linear_regression_model = LinearRegression(fit_intercept=True).fit(X=train_linear_df['X']`, but maybe I am missing something. The same pattern is in logistic regression section.
> * When you finish your work on notebook, I would suggest to restart kernel and run all cells, so the numbers of cells are ordered. It is also a check for you that notebook can be re-run by other user without any errors.
> * In the "EXERCISES 1B" section, there is written the following: `1b) Use sklearn's linear regression model on the non-zero elements of the 'age' feature to predict the zero-elements of the 'age' feature.` However, I think you meant something like this: `1b) Train sklearn's linear regression model on the non-zero elements of the 'age' feature to predict the missing values of the 'age' feature.`
> * Links section in PR description does not make sense to me. Please, link the issue you were working on.

I was also thinking about Linda's powers. However, it would be hard to do edits :/

**Comment by niekstrv** [comment]
> 

@jardabezdek fixed, except for the point with the reshaping of the intput for linear/logistic regression. This is wat SKlearn forces me to do whern the input is 1d. I am not aware of any other correction except for the on in the NB

**Comment by jardabezdek** [comment]
> > 
> 
> @jardabezdek fixed, except for the point with the reshaping of the intput for linear/logistic regression. This is wat SKlearn forces me to do whern the input is 1d. I am not aware of any other correction except for the on in the NB

@niekstrv Oh, I see. The reason is that the `scikit-learn` model expects X to be a 2D array. It means you can simplify the code from this:

```
linear_regression_model = (
    LinearRegression(fit_intercept=True)
    .fit(
        X=train_linear_df['X'].values.reshape(-1, 1), 
        y=train_linear_df['y']
    )
)
```

to this (add `'X'` into double brackets):

```
linear_regression_model = (
    LinearRegression(fit_intercept=True)
    .fit(
        X=train_linear_df[['X']], 
        y=train_linear_df['y']
    )
)
```
